### PR TITLE
Add `test_connection` method to AzureCosmosDBHook

### PR DESCRIPTION
**Pull request by phanikumv**
This PR enables the Test Connection button on the airflow UI for the connection type Azure CosmosDB.

cc @kaxil 

Screenshot 2022-07-13 at 11 27 15 AM

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/microsoft/azure/hooks/cosmos.py [hunk 1]
```diff
@@ -320,6 +320,25 @@ def get_documents(
...
+            # connection.
+            next(iter(self.get_conn().list_databases()))
+            return success
+        except StopIteration:
+            # If the iterator returned is empty it should still be considered a successful connection since
+            # it's possible to create a Cosmos connection via the ``AzureCosmosDBHook`` and no database could
+            # legitimately exist yet.
+            return success
+        except Exception as e:
+            return False, str(e)
```
Comment (uranusjr) [inline airflow/providers/microsoft/azure/hooks/cosmos.py]: If we’re ignoring both the content and the StopIteration case, it’s much easier to do

```python
try:
    next(iter(self.get_conn().list_databases()), None)
except Exception as e:
    return False, str(e)
return (True, "Successfully connected to Azure Cosmos.")

Comment (phanikumv) [reply airflow/providers/microsoft/azure/hooks/cosmos.py]: `StopIteration` will be raised if there is no database created . Am I understanding it correctly that the `None` will cover for that scenario @uranusjr ?

```
>>> next(iter([]), None)
>>> next(iter([]))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
StopIteration
```

Comment (uranusjr) [reply airflow/providers/microsoft/azure/hooks/cosmos.py]: Yes

https://docs.python.org/3/library/functions.html#next

Comment (phanikumv) [reply airflow/providers/microsoft/azure/hooks/cosmos.py]: Thanks ! Changed the code as per suggestion.

**Review by uranusjr — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

**Pull request merged**